### PR TITLE
Followed Sites: Replace success / failure alert with Notice

### LINKED ISSUE #8514: Better error message when trying to follow a site which has already been followed.
### Expected behavior
Unobtrusive alert saying that a site has already been followed.

### Actual behavior
Scary modal error message.

### Steps to reproduce the behavior
Try to follow a site that has already been followed.

##### Tested on [device], iOS [version], WPiOS [version]
iOS, current.

Reported by @m.

image uploaded from ios

### PR DESCRIPTION
**Pull request by frosty**
Fixes #8514 

This PR replaces the alert that's displayed when you attempt to follow an already-followed site with a Notice at the bottom of the screen. It also replaces the success HUD when you follow a site.

**To test:**

* Go to Reader > Followed Sites and attempt to follow a site that you've already followed. Check that the notice is displayed at the bottom of the screen.
* On the same screen, follow a new site. Check that a different notice is displayed.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift [hunk 1]
```diff
@@ -202,25 +203,25 @@ class ReaderFollowedSitesViewController: UIViewController, UIViewControllerResto
...
 
-        let generator = UINotificationFeedbackGenerator()
-        generator.prepare()
-
         let service = ReaderSiteService(managedObjectContext: managedObjectContext())
         service.followSite(by: url, success: { [weak self] in
-            let success = NSLocalizedString("Followed", comment: "User followed a site.")
-            SVProgressHUD.showDismissibleSuccess(withStatus: success)
-            generator.notificationOccurred(.success)
+            SVProgressHUD.dismiss()
```
Comment (elibud) [inline WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift]: Why is this line needed?

**Comment by folletto** [comment]
Can you post a before+after gif or screenshot?

**Comment by elibud** [comment]
Sure, here it is. There's haptic feedback, but that can be disabled and I don't find it so straight forward to interpret.

| Before  | After |
| ------------- | ------------- |
| beforesuccess  | aftersuccess  |
|beforefailure  |afterfailure  |

**Comment by folletto** [comment]
I agree with Eli's comment, but I'd still proceed as it's an improvement in standardization and flexibility. I also hope that by having the site actually added to the list it's still clear enough and easy enough to cross check.

If possible I'd maybe make two tweaks...

:one: Add actions to both the messages:

> [ Open site ]

That would open the Reader view of the single site... if it's not too complicated. :) 

:two:  Change the text in case of failure to:

> You are already following it.
> $sitename

...and maybe scroll the list to it?

File: WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift (hunk 1 above)
Comment (frosty) [reply WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift]: You're right, this and the `import` can be removed 👍

**Comment by frosty** [comment]
@folletto Thanks for the suggestions! Unfortunately:

* I've looked through the code and I don't think there's any easy way to add an Open Site button. The way the web service currently works for following sites, we don't get the new followed site object passed back so there's not really any way of knowing which site is new. In the case of failures, we wouldn't have a reader site object at all, and there's no way of opening a Reader view for an arbitrary URL.
* This also means we can't easily scroll to the site in the list.

I could add an Open Site button and simply open the site in a web view, but I suspect we don't really want to take folks out of viewing things in the reader?

I can change the failure text. Could I counter your suggestion with simply "You are already following $sitename."?

**Comment by elibud** [comment]
@frosty OK lets ship this after removing the unnecessary code.

> I can change the failure text. Could I counter your suggestion with simply "You are already following $sitename."?

Since following may fail for any other reason (for example no internet connection) I don't think we should make up our own error, what do you think?

**Comment by frosty** [comment]
@elibud The "You are already following that site" message we already have is our own error description that comes from `ReaderSiteService` if we detect that that's the reason for the failure. I can modify that a little if necessary to say "You are already following $sitename".

**Comment by elibud** [comment]
oh great, then yes please modify it :) I though the error was coming from the backend 🤦‍♀️

**Comment by frosty** [comment]
@elibud Want to take another look? I updated the unfollow UI too.

**Comment by folletto** [comment]
> Unfortunately

It's ok. Thanks for looking into that. :)

> I can change the failure text. Could I counter your suggestion with simply "You are already following $sitename."?

Sure. I was just trying to identify a way to say it that kept the structure of two rows where the second was the site. :)

**Pull request merged**